person.rb

```diff
@@ -2,36 +2,94 @@
 require_relative './base_decorator'
 require_relative './capitalize_decorator'
 require_relative './trimmer_decorator'
+require_relative './rental'
 
 class Person < Nameable
   attr_accessor :id, :name, :age, :parent_permission
 
-  def initialize(age, name, id, parent_permission: true)
+  def initialize(age, id, parent_permission: true)
     super()
     @id = id
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
+  end
+
+  def add_rental(book, date)
+    Rental.new(date, self, book)
+  end
+
+  def correct_name
+    @name
   end
 
   def can_use_services?
     of_age? || @parent_permission
   end
 
-  def of_age?
-    @age >= 18
+  def self.create_person(persons)
+    print 'Do you want to create a student (1) or teacher (2) [Input a number]: '
+    option = gets.chomp.to_i
+
+    case option
+
+    when 1
+
+      person = create_student
+    when 2
+
+      person = create_teacher
+    else
+      puts 'Invalid input. Kindly type 1 or 2'
+    end
+    persons << person
+
+    # print "#{person.class} created successfully\n"
   end
 
-  def correct_name
-    @name
+  def self.create_student
+    print 'Classroom: '
+    classroom = gets.chomp
+
+    print 'Age: '
+    age = Interger(gets.chomp)
```

Comment:
- [ ] I got this error while trying to create a student. It seems you misspelled Integer 👇🏽 

![image](https://user-images.githubusercontent.com/63473584/198849042-1648f9dc-3d8d-4e18-8f48-578c8c830b4a.png)